sync/src/main/java/tech/pegasys/teku/sync/PeerSyncBlockRequest.java

```diff
@@ -37,7 +37,15 @@ public PeerSyncBlockRequest(
 
   @Override
   public SafeFuture<?> onResponse(final SignedBeaconBlock response) {
-    slotOfLastBlock = Optional.of(response.getSlot());
+    UnsignedLong newBlockSlot = response.getSlot();
+    if (maybeSlotOfLastBlock.isPresent()) {
+      UnsignedLong slotOfLastBlock = maybeSlotOfLastBlock.get();
+      if (newBlockSlot.compareTo(slotOfLastBlock) <= 0) {
+        throw new OutOfOrderException(slotOfLastBlock, newBlockSlot);
+      }
```

Comment:
Would also be a good idea to check that parent root matches up with the previous block root if the step == 1